### PR TITLE
Update to OTel 0.14

### PR DESCRIPTION
**Pull request by anuraaga**
*Issue #, if available:*

*Description of changes:*
Had to work around issue that prevents SPI configuration to work until 0.15.0. Fix is in upstream in https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2077 but that has breaking changes to configuration options which we can't push out in a patch. This workaround is tame enough.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: awsagentprovider/src/main/java/software/amazon/opentelemetry/javaagent/providers/AwsTracerProviderFactory.java [hunk 1]
```diff
@@ -15,13 +15,12 @@
...
-import io.opentelemetry.api.OpenTelemetry;
-import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.api.trace.TracerProvider;
 import io.opentelemetry.sdk.extension.aws.trace.AwsXrayIdGenerator;
 import io.opentelemetry.sdk.trace.SdkTracerProvider;
-import io.opentelemetry.spi.OpenTelemetryFactory;
+import io.opentelemetry.spi.trace.TracerProviderFactory;
 
-public class AwsOpenTelemetryFactory implements OpenTelemetryFactory {
+public class AwsTracerProviderFactory implements TracerProviderFactory {
```
Comment (anuraaga) [inline awsagentprovider/src/main/java/software/amazon/opentelemetry/javaagent/providers/AwsTracerProviderFactory.java:23]: Unnecessary changes, these will change again in 0.15.0 to `implements TracerProviderConfigurer`

File: awsagentprovider/src/main/java/io/opentelemetry/sdk/trace/RandomIdGenerator.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  @Override
+  public String generateSpanId() {
+    // Inline implementation since AwsXrayIdGenerator would delegate back to RandomIdGenerator
+    // causing an infinite loop.
+    long id;
+    ThreadLocalRandom random = ThreadLocalRandom.current();
+    do {
+      id = random.nextLong();
+    } while (id == INVALID_ID);
```
Comment (willarmiros) [inline awsagentprovider/src/main/java/io/opentelemetry/sdk/trace/RandomIdGenerator.java:38]: jw, why a `do..while` loop here?

**Review by willarmiros — APPROVED** [review]
LGTM

File: awsagentprovider/src/main/java/io/opentelemetry/sdk/trace/RandomIdGenerator.java (hunk 2 above)
Comment (anuraaga) [reply awsagentprovider/src/main/java/io/opentelemetry/sdk/trace/RandomIdGenerator.java:38]: Copied this from upstream - it's just to avoid the random being `0` which is invalid, even though the probability of that is extremely low.
